invoiceregistry_test.go

```diff
@@ -496,11 +496,93 @@ func TestOverpayment(t *testing.T) {
 	cancelAcceptUpdates()
 }
 
+func TestMPPValidity(t *testing.T) {
+	defer test.Timeout()()
+	t.Parallel()
+
+	c := newRegistryTestContext(t, optAutoSettle(true))
+
+	resolved := make(chan HtlcResolution)
+
+	notifyHtlc := func(id uint64, rHash lntypes.Hash,
+		amtPaid lnwire.MilliSatoshi, curHeight int32,
+		payload *testPayload) {
+
+		c.registry.NotifyExitHopHtlc(&registryHtlc{
+			rHash:         rHash,
+			amtPaid:       amtPaid,
+			expiry:        100,
+			currentHeight: curHeight,
+			circuitKey: types.CircuitKey{
+				HtlcID: id,
+			},
+			resolve: func(r HtlcResolution) {
+				resolved <- r
+			},
+			payload: payload,
+		})
+	}
+
+	// Create invoice that's expired.
+	invoice, preimage := c.createInvoice(1, time.Hour)
+	rHash := preimage.Hash()
+
+	testAmt := lnwire.MilliSatoshi(c.testAmt)
+
+	// Test HTLC with no MPP record.
+	notifyHtlc(1, rHash, testAmt, 0, &testPayload{
+		amt:     testAmt,
+		payAddr: invoice.PaymentAddr,
+		nullMpp: true,
+	})
+
+	c.checkHtlcFailed(<-resolved, ResultHtlcInvoiceTypeMismatch)
+
+	// Test MPP with total too low.
+	notifyHtlc(2, rHash, testAmt, 0, &testPayload{
+		amt:     lnwire.MilliSatoshi(0),
+		payAddr: invoice.PaymentAddr,
+	})
+
+	c.checkHtlcFailed(<-resolved, ResultHtlcSetTotalTooLow)
+
+	// Test MPP with hash mismatching the invoice.
+	notifyHtlc(3, lntypes.ZeroHash, testAmt, 0, &testPayload{
+		amt:     testAmt,
+		payAddr: invoice.PaymentAddr,
+	})
+
+	// TODO(aakselrod): should this be ResultAddressMismatch instead?
```

Comment:
Yes, I think it makes sense to differentiate here.

Reply:
Fixed!